itests/kit/evm.go

```diff
@@ -83,8 +83,11 @@ func (e *EVM) DeployContract(ctx context.Context, sender address.Address, byteco
 
 	return result
 }
+func (e *EVM) DeployContract(ctx context.Context, sender address.Address, bytecode []byte) eam.CreateReturn {
+	return e.DeployContractValue(ctx, sender, bytecode, big.Zero())
+}
 
-func (e *EVM) DeployContractFromFilename(ctx context.Context, binFilename string) (address.Address, address.Address) {
+func (e *EVM) DeployContractFromFilenameValue(ctx context.Context, binFilename string, value big.Int) (address.Address, address.Address) {
```

Comment:
I think this could be named `DeployContractFromFileWithValue`.

I think all the functions here should end with `WithValue` instead of just `Value`. I think the latter makes sense to us, but is probably ambiguous to anyone looking at this code fresh.

Reply:
There are a lot of variants here -- looks like an API cartesian product situation. This is where the functional options pattern comes in handy: https://golang.cafe/blog/golang-functional-options-pattern.html.

Reply:
Thanks @raulk @geoff-vball ! This feedback is super helpful! I have updated the PR to use WithValue